core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/HDFSFileMoveAction.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.action;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.action.Action;
+import co.cask.cdap.etl.api.action.ActionContext;
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.PathFilter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * Action that moves file(s) within HDFS in the same cluster.
+ * A user must specify file/directory path and destination file/directory path
+ * Optionals include fileRegex
+ */
+@Plugin(type = Action.PLUGIN_TYPE)
+@Name("HDFSFileMoveAction")
+@Description("Action to move files within HDFS")
+public class HDFSFileMoveAction extends Action {
+  private static final Logger LOG = LoggerFactory.getLogger(HDFSFileMoveAction.class);
+
+  private HDFSActionConfig config;
+
+  public HDFSFileMoveAction(HDFSActionConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    Path source = new Path(config.sourcePath);
+    new File(config.destPath).mkdirs();
```

Comment:
this doesn't work. File is java's api for local files. Its also incorrect logic in case the the destination is a file and not a directory. Please add a test case for this. Don't just add code and assume it works. You need to test it out.